linux/Platform.c

```diff
@@ -555,23 +557,10 @@ bool Platform_getDiskIO(DiskIOData* data) {
          if (String_startsWith(diskname, "zram"))
             continue;
 
-         /* only count root disks, e.g. do not count IO from sda and sda1 twice */
-         if ((diskname[0] == 's' || diskname[0] == 'h')
-             && diskname[1] == 'd'
-             && isalpha((unsigned char)diskname[2])
-             && isdigit((unsigned char)diskname[3]))
-            continue;
-
-         /* only count root disks, e.g. do not count IO from mmcblk0 and mmcblk0p1 twice */
-         if (diskname[0] == 'm'
-             && diskname[1] == 'm'
-             && diskname[2] == 'c'
-             && diskname[3] == 'b'
-             && diskname[4] == 'l'
-             && diskname[5] == 'k'
-             && isdigit((unsigned char)diskname[6])
-             && diskname[7] == 'p')
-            continue;
+         if (lastTopDisk[0] && String_startsWith(diskname, lastTopDisk))
```

Comment:
Document the implicit requirement of disks being reported before their partitions in `/proc/diskstats`.